read-cache.c

```diff
@@ -1899,7 +1899,7 @@ static size_t estimate_cache_size(size_t ondisk_size, unsigned int entries)
 	/*
 	 * Account for potential alignment differences.
 	 */
-	per_entry += align_padding_size(sizeof(struct cache_entry), -sizeof(struct ondisk_cache_entry));
+	per_entry += align_padding_size(per_entry, 0);
```

Comment:
The subtleties of this makes me nervous, but I'm not sure the original was any better, TBH.

Reply:
Right. I think with the addition to the commit message to which @PhilipOakley nudged me, the new version is substantially better ;-)